next/13/replace-api-routes/index.ts

```diff
@@ -0,0 +1,329 @@
+import tsmorph, {
+	ArrowFunction,
+	FunctionDeclaration,
+	FunctionExpression,
+	Node,
+	SyntaxKind,
+	SourceFile,
+	CallExpression,
+} from 'ts-morph';
+
+import type {
+	Repomod,
+	UnifiedFileSystem,
+} from '@intuita-inc/repomod-engine-api';
+
+import { posix } from 'node:path';
+import { ParsedPath } from 'path/posix';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type Dependencies = Readonly<{
+	tsmorph: typeof tsmorph;
+	unifiedFileSystem: UnifiedFileSystem;
+}>;
+
+type Handler = ArrowFunction | FunctionExpression | FunctionDeclaration;
+
+const getNewDirectoryName = ({ dir, name }: ParsedPath) => {
+	const directoryNameSegments = dir.split(posix.sep);
+
+	const newDirectoryNameSegments = directoryNameSegments.map((segment) =>
+		segment === 'pages' ? 'app' : segment,
+	);
+
+	if (name !== 'index') {
+		newDirectoryNameSegments.push(name);
+	}
+
+	return newDirectoryNameSegments.join(posix.sep);
+};
+
+const findAPIRouteHandler = (sourceFile: SourceFile): Handler | null => {
+	const defaultExportedFunctionDeclaration = sourceFile
+		.getFunctions()
+		.find((f) => f.isDefaultExport());
+
+	if (defaultExportedFunctionDeclaration !== undefined) {
+		return defaultExportedFunctionDeclaration;
+	}
+
+	const exportAssignment = sourceFile
+		.getStatements()
+		.find((s) => Node.isExportAssignment(s));
+
+	const declarations =
+		exportAssignment
+			?.getFirstDescendantByKind(SyntaxKind.Identifier)
+			?.getSymbol()
+			?.getDeclarations() ?? [];
+
+	let handler:
+		| ArrowFunction
+		| FunctionExpression
+		| FunctionDeclaration
+		| undefined;
+
+	declarations.forEach((d) => {
+		if (Node.isVariableDeclaration(d)) {
+			const initializer = d?.getInitializer();
+
+			if (
+				Node.isArrowFunction(initializer) ||
+				Node.isFunctionExpression(initializer)
+			) {
+				handler = initializer;
+				return;
+			}
+		}
+
+		if (Node.isFunctionDeclaration(d)) {
+			handler = d;
+		}
+	});
+
+	return handler ?? null;
+};
+
+const getPositionAfterImports = (sourceFile: SourceFile): number => {
+	const lastImportDeclaration =
+		sourceFile.getLastChildByKind(SyntaxKind.ImportDeclaration) ?? null;
+
+	return (lastImportDeclaration?.getChildIndex() ?? 0) + 1;
+};
+
+const HTTP_METHODS = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH'] as const;
+
+export type HTTPMethod = (typeof HTTP_METHODS)[number];
+
+const RESPONSE_INIT_FIELDS = ['headers', 'status', 'statusText'] as const;
+
+type ResponseInitParam = (typeof RESPONSE_INIT_FIELDS)[number];
+type ResponseInit = Partial<{ [k in ResponseInitParam]: unknown }>;
+
+const unquotify = (input: string): string => {
```

Comment:
what was the reason for that function's existence?

Reply:
sometimes `getText` returns a string with extra quotes like `'next/server'` or `"next/server"`, so i have to remove them

Reply:
fixed